### PR TITLE
checkpoint: explicitly load checkpoint manifests

### PR DESCRIPTION
**Pull request by xiang90**
@aaronlevy 

Now we load the existing manifests, and only try to install the manifest if checkpoint exists. The previously logic try to create static pod blindly. 

Last change for the preparation of the final code logic restructure. With the disk manifest preloading, the checkpointer knows exactly what it expects to recreate if not exists.

### CODE REVIEW

**Comment by aaronlevy** [comment]
test checkpoint code please

**Comment by aaronlevy** [comment]
Just fyi the above is to trigger the jenkins job. Code-wise this lgtm

**Comment by xiang90** [comment]
@aaronlevy I can see the bootkube-dev turns green. Does this mean the test get passed? cc @pbx0

**Comment by aaronlevy** [comment]
test checkpoint code please

**Comment by xiang90** [comment]
@aaronlevy :( It is not.

**Comment by xiang90** [comment]
test checkpoint code please

**Comment by xiang90** [comment]
test checkpoint code please

**Comment by xiang90** [comment]
test checkpoint code please

File: cmd/checkpoint/main.go [hunk 1]
```diff
@@ -62,17 +69,18 @@ func run(actualPodName, tempPodName, namespace string) {
...
 			checkpointPod := createCheckpointPod(podList, actualPodName, namespace)
 			convertSecretsToVolumeMounts(client, &checkpointPod)
 			writeManifest(checkpointPod, tempPodName)
-			glog.Infof("finished creating temp pod %v manifest at %s\n", tempPodName, checkpointManifest(tempPodName))
+			checkpoints[checkpointManifest(tempPodName)] = struct{}{}
+			glog.Infof("finished creating checkpoint pod %v manifest at %s\n", tempPodName, checkpointManifest(tempPodName))
 
-		default:
-			glog.Info("no actual pod running, installing temp pod static manifest")
+		case checkpointed:
```
Comment (xiang90) [inline cmd/checkpoint/main.go]: You are right. Fixed.

**Comment by aaronlevy** [comment]
coreosbot test checkpoint code please

**Comment by aaronlevy** [comment]
coreosbot test checkpoint code

**Pull request merged**